src/JustSaying/Messaging/Monitoring/IMessageMonitor.cs

```diff
@@ -3,16 +3,69 @@
 
 namespace JustSaying.Messaging.Monitoring
 {
+    /// <summary>
+    /// Provides a mechanism for monitoring when various message actions are
+    /// performed in JustSaying. This is typically used for metrics gathering.
+    /// </summary>
     public interface IMessageMonitor
     {
+        /// <summary>
+        /// Monitors when a specific message type threw an exception while being handled
+        /// by a subscriber.
+        /// </summary>
+        /// <param name="messageType">The type of the message that was being handled.</param>
         void HandleException(Type messageType);
+
+        /// <summary>
+        /// Monitors when a particular message that threw an exception while being handled
+        /// by a subscriber.
+        /// </summary>
+        /// <param name="ex">The exception that was thrown.</param>
+        /// <param name="message">The message being handled when the exception was thrown.</param>
         void HandleError(Exception ex, Message message);
+
+        /// <summary>
+        /// Monitors the length of time a message took to be handled by a subscriber.
+        /// </summary>
+        /// <param name="duration">The length of time taken.</param>
         void HandleTime(TimeSpan duration);
+
+        /// <summary>
+        /// Monitors when a message was not successfully published after waiting for
+        /// the <see cref="IPublishConfiguration.PublishFailureReAttempts">configured amount of retries</see>.
+        /// </summary>
         void IssuePublishingMessage();
+
+        /// <summary>
+        /// Monitors when each message is handled by a subscriber.
+        /// </summary>
+        /// <param name="message">The message that was handled.</param>
         void Handled(JustSaying.Models.Message message);
+
+        /// <summary>
+        /// Monitors each time a message took over 1ms to be written to a ChannelWriter.
```

Comment:
We perhaps don't want to be too specific about the implementation details of how things are throttled. Would it be enough to just say something like `Called when a message pipeline is full, and a receiver must wait until there is room to continue downloading messages`?